tcms/testcases/views.py

```diff
@@ -771,33 +800,39 @@ def get(request, case_id):
     """Get the case content"""
     # Get the case
     try:
-        tc = TestCase.objects.select_related(
+        test_case = TestCase.objects.select_related(
             'author', 'default_tester',
             'category', 'category',
             'priority', 'case_status').get(case_id=case_id)
     except ObjectDoesNotExist:
         raise Http404
 
     # Get the test plans
-    tps = tc.plan.select_related('author', 'product', 'type').all()
+    tps = test_case.plan.select_related('author', 'product', 'type').all()
 
     try:
-        tp = tps.get(pk=request.GET.get('from_plan', 0))
+        test_plan = tps.get(pk=request.GET.get('from_plan', 0))
     except (TestPlan.DoesNotExist, ValueError):
         # ValueError is raised when from_plan is empty string
         # not viewing TC from a Plan or specified Plan does not exist (e.g. broken link)
-        tp = None
+        test_plan = None
 
     # Get the test case runs
-    tcrs = tc.case_run.select_related(
+    tcrs = test_case.case_run.select_related(
         'run', 'tested_by',
         'assignee', 'case',
         'case', 'case_run_status').order_by('run__plan')
     runs_ordered_by_plan = itertools.groupby(tcrs, lambda t: t.run.plan)
     # FIXME: Just don't know why Django template does not evaluate a generator,
     # and had to evaluate the groupby generator manually like below.
-    runs_ordered_by_plan = [(k, list(v)) for k, v in runs_ordered_by_plan]
-    case_run_plans = [k for k, v in runs_ordered_by_plan]
+    runs_ordered_by_plan = []
+    for key, value in runs_ordered_by_plan:
+        runs_ordered_by_plan.append((key, list(value)))
```

Comment:
This is the same error as seen above when working with the bugs. L828 essentially makes the list of runs empty and the loops below will never iterate!

This breaks the UI, TestCase get view, Case Runs tab is now empty! You will have to create a test for that (in a 2nd PR once the 1st one is done) and fix it. 

Note: ^^^ you have to work in smaller chunks next time and learn to also manually test and inspect the application when you introduce changes.